src/main/java/net/minecraftforge/common/command/SelectorHandler.java

```
@@ -0,0 +1,102 @@
+ * <b>Note:</b>  You should generally override all three methods to produce consistent results<br>
+ * <b>Note:</b> These methods are called on both client and server<br>
+ * <b>Note:</b>  For compatibility reasons, you should not break the key=value pattern
+ */
+public class SelectorHandler
+{
+    public <T extends Entity> List<T> matchEntities(ICommandSender sender, String token, Class<? extends T> targetClass) throws CommandException
+    {
+        return EntitySelector.matchEntitiesDefault(sender, token, targetClass);
+    }
+
+    /**
+     * This should raise an {@link net.minecraftforge.event.EntitySelectorEvent EntitySelectorEvent}
+     */
+    public boolean matchesMultiplePlayers(String selectorStr) throws CommandException
+    {
+        return EntitySelector.matchesMultiplePlayersDefault(selectorStr);
+    }
+
+    public boolean hasArguments(String selectorStr)
+    {
+        return EntitySelector.hasArgumentsDefault(selectorStr);
+    }
+
+    private static SelectorHandler handler = new SelectorHandler();
+    private static boolean isCustom = false;
+    private static String modifyingMod = "Minecraft";
+
+    /**
+     * Allows you to override the behavior of EnitySelector by setting this handler to a custom implementation<br>
+     * This overrides any previous handlers set<br>
+     * Always use {@link net.minecraftforge.event.EntitySelectorEvent EntitySelectorEvent} if possible
+     */
+    public static void setHandler(SelectorHandler handler)
+    {
+        if (isCustom)
+            FMLLog.bigWarning("SelectorHandler is already overridden" + (modifyingMod == null ? "" : " by '" + modifyingMod + "'"));
+
+        isCustom = true;
+        ModContainer container = Loader.instance().activeModContainer();
+
+        if (container == null)
+            FMLLog.bigWarning("SelectorHandler is overridden by an unknown mod. Overriding should be done during mod loading");
```

This one should throw an exception if it's important to know the modifying mod.